libass/ass.c

```diff
@@ -1330,6 +1341,7 @@ void ass_process_chunk(ASS_Track *track, const char *data, int size,
 
         event->Start = timecode;
         event->Duration = duration;
+        track->parser_priv->gc_next_undisplay_ts = FFMIN(track->parser_priv->gc_next_undisplay_ts, event->Start + event->Duration);
```

Comment:
The start and duration fix-ups above prevent this line to be in `process_event_tail`, preventing a clean implementation.

Reply:
We could move time assigns here before `process_event_tail`, but in the pathological case where
- custom format lines are used
- `Start` and/or `End` appear (again?) as or after the fourth element
- one of these value differs from packet timestamps

However this case is arguably already somewhat broken and might not be worth worrying about